OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -123,9 +125,16 @@ protected override void OnFirstRun(Actor self)
 			// TODO: Change this to BlockedByActor.Stationary after improving the local avoidance behaviour
 			foreach (var check in PathSearchOrder)
 			{
-				path = EvalPath(check);
+				var (pathing, destinationIsStartpoint) = EvalPath(check);
+				path = pathing;
+
 				if (path.Count > 0)
 					return;
+
+				// If the terrain is not even reachable, suggest parent activity to give up
+				// HACK: see "MoveAdjacentTo.cs" -> "CalculatePathToTarget(...)" comments on why we need "destinationIsStartpoint"
+				if (check == BlockedByActor.None && !destinationIsStartpoint)
+					mobile.MoveResult = MoveResult.MovementStuck;
```

Comment:
Don't set the MoveResult here. Instead, we should set the MoveResult in the tick method.

Whenever Tick is about to return true, the activity is complete. Therefore we should ensure the MoveResult is set to one of the complete states just before we return true. If we return false from Tick, then the MoveResult should still be `InProgress`.

I am hoping we can make the `destinationIsStartpoint` cleaner as well. Hopefully we can do some sort of check that works out whether we reached the destination, so we don't need to track `destinationIsStartpoint`.